apps/aefate/src/aefa_chain_api.erl

```diff
@@ -250,6 +285,38 @@ account_balance(Pubkey, #state{primop_state = PState,
             end
     end.
 
+-spec next_nonce(pubkey(), state()) -> 'error' |
+                                       {'ok', integer(), state()}.
+next_nonce(Pubkey, #state{primop_state = PState0,
+                          onchain_primop_state = Onchain} = S) ->
+    case aeprimop_state:find_account(Pubkey, PState0) of
+        {Account, PState1} ->
+            Nonce = aec_accounts:nonce(Account) + 1,
+            case eval_primops(
+                   [aeprimop:force_inc_account_nonce_op(Pubkey, Nonce)],
+                   S#state{primop_state = PState1}) of
+                {ok, S1} -> {ok, Nonce, S1};
+                Err -> Err
+            end;
+        none when Onchain =:= none ->
+            error;
+        none ->
+            %% Try to find this onchain as well.
```

Comment:
I have to say I'm quite unsure if we should look onchain. I though that state channels contracts can't interact in statefull way with onchain

Reply:
But this is not a stateful interaction from the perspective of chain, is it?